external-crates/move/crates/move-compiler/src/cfgir/translate.rs

```diff
@@ -983,17 +986,22 @@ fn destructure_tuple<T, U>((fst, snd): &(T, U)) -> (&T, &U) {
 // Visitors
 //**************************************************************************************************
 
-fn visit_program(context: &mut Context, prog: &G::Program) {
-    if context.env.visitors().abs_int.is_empty() {
+fn visit_program(context: &mut Context, prog: &mut G::Program) {
+    if context.env.visitors().abs_int.is_empty() && context.env.visitors().cfgir.is_empty() {
         return;
     }
 
     for (mident, mdef) in prog.modules.key_cloned_iter() {
-        visit_module(context, prog, mident, mdef)
+        absint_visit_module(context, prog, mident, mdef)
+    }
+
+    for visitor in &context.env.visitors().cfgir {
+        let mut v = visitor.borrow_mut();
+        v.visit(context.env, prog)
     }
 }
 
-fn visit_module(
+fn absint_visit_module(
```

Comment:
Could we rewrite this to use the new visitor?

Reply:
Oh thats fun. Stuffing all of the absint visitors in a visitor?